bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -234,9 +237,9 @@ public void setFilters(@Nullable String[] filters) {
      *
      * @return Ignore classes
      */
-    @Nullable
-    public String[] getIgnoreClasses() {
-        return ignoreClasses;
+    @NonNull
+    public Collection<String> getIgnoreClasses() {
+        return Collections.unmodifiableSet(ignoreClasses);
```

Comment:
why is this made immutable upon `get()`? Why not make it an unmodifiable set when it is received in `setIgnoreClasses()` ? Same goes for `notifyReleaseStages` and `projectPackages`

Reply:
Java doesn't really have a truly immutable collection. `Collections.unmodifiableSet` provides a view of an existing collection to an end-user, that will throw an exception when operations such as element insertion/deletion are attempted. Either approach would be valid - arguably doing it in the setter would be better as it would prevent `Configuration` mutating the value, but #555 makes a truly immutable copy of this property anyway.